pkg/framework/host.go

```diff
@@ -207,28 +207,60 @@ func (h *Host) InstallResource(name, values string, conditions ...func() error)
 }
 
 func (h *Host) InstallCertResource() error {
-	operation := func() error {
-		// NOTE we ignore errors here because we cannot get really useful error
-		// handling done. This here should anyway only be a quick fix until we use
-		// the helm client lib. Then error handling will be better.
-		HelmCmd("delete --purge cert-resource-lab")
+	{
+		log.Println("level", "debug", "message", "installing cert resource chart")
+
+		o := func() error {
+			// NOTE we ignore errors here because we cannot get really useful error
+			// handling done. This here should anyway only be a quick fix until we use
+			// the helm client lib. Then error handling will be better.
+			HelmCmd("delete --purge cert-resource-lab")
+
+			err := HelmCmd("registry install quay.io/giantswarm/cert-resource-lab-chart:stable -- -n cert-resource-lab --set commonDomain=${COMMON_DOMAIN_GUEST} --set clusterName=${CLUSTER_NAME}")
+			if err != nil {
+				return microerror.Mask(err)
+			}
 
-		err := HelmCmd("registry install quay.io/giantswarm/cert-resource-lab-chart:stable -- -n cert-resource-lab --set commonDomain=${COMMON_DOMAIN_GUEST} --set clusterName=${CLUSTER_NAME}")
+			return nil
+		}
+		b := NewExponentialBackoff(ShortMaxWait, ShortMaxInterval)
+		n := newNotify("cert-resource-lab-chart install")
+		err := backoff.RetryNotify(o, b, n)
 		if err != nil {
 			return microerror.Mask(err)
 		}
 
-		return nil
+		log.Println("level", "debug", "message", "installed cert resource chart")
 	}
-	notify := newNotify("cert-resource-lab-chart install")
-	err := backoff.RetryNotify(operation, h.backoff, notify)
-	if err != nil {
-		return microerror.Mask(err)
+
+	{
+		log.Println("level", "debug", "message", "waiting for k8s secret to be there")
+
+		o := func() error {
```

Comment:
Why not using the h.secret() function here? It is doing the same thing.

```
n := fmt.Sprintf("%s-api", os.Getenv("CLUSTER_NAME"))
o := h.secret("default", n)
```

Reply:
Because I want to refactor the code and get the mental model straight in my head instead of using the old stuff that is there. When it turns out we want to have the operation being separated and reusable I will be glad to provide a PR to do this. But I need to see this and get a feel for the framework first.